File: C++/Medium/Accepted/minimized-maximum-of-products-distributed-to-any-store_9.cpp
class Solution {
public:
    int minimizedMaximum(int n, vector<int>& quantities) {
        
        int m = quantities.size();
        int l = 1, r = 1e5;
        while(l < r){
            int mid = (l + r) / 2;
            int sum = 0;
            for(int i = 0; i < m; i++){
                sum += (quantities[i] + mid - 1) / mid;
            }
            if(sum <= n){
                r = mid;
            }else{
                l = mid + 1;
            }
        }
        return l;
    }
};

File: Java/Medium/RuntimeError/minimized-maximum-of-products-distributed-to-any-store_9.java
class Solution {
    public int minimizedMaximum(int n, int[] quantities) {
        
        int sum = 0;
        for(int q: quantities) sum += q;
        int l = 0, r = sum;
        while(l < r) {
            int mid = (l + r) / 2;
            if(check(quantities, mid, n)) {
                r = mid;
            } else {
                l = mid + 1;
            }
        }
        return l;
    }
    
    private boolean check(int[] quantities, int x, int n) {
        int sum = 0;
        for(int q: quantities) {
            sum += (q + x - 1) / x;
        }
        return sum <= n;
    }
}